### ISSUE TITLE
--audio-width not translating quite right to mixdown

### ISSUE THREAD

**Issue by Thor263** [donmelton/video_transcoding#276]
Hi.

I'm changing order of audio tracks from the original video and am using `--audio-width` to set the main track (DTS-HD surround track 2 on original) to surround eac3, the new second track (DTS-HD stereo track 1 on original) to stereo aac, the 3rd track (DTS-HD mono track 3 on original) to mono aac and passthrough the 4th track (track 12 on original) as AC3. However, the expected mixdown for the second track ends up as surround (no mixdown specified in HandBrakeCLI).

Here's the dry-run using transcode-video 0.25.1, under macOS 10.14.4:
```
transcode-video --dry-run --avbr --target big --max-width 1920 --max-height 1080 --main-audio 2 --add-audio 1  --add-audio 3  --add-audio 12 --ac3-encoder eac3 --audio-width 2=surround --audio-width 1=stereo --keep-ac3-stereo --crop 264:264:2:0 /path/to/Movie.mkv

HandBrakeCLI --input=/path/to/Movie.mkv --output=Movie.mkv --markers --encoder=x264 --crop=264:264:2:0 --loose-anamorphic --encoder-profile=high --encoder-level=4.0 --vb=8000 --audio=2,1,3,12 --aencoder=eac3,ca_aac,ca_aac,copy --ab=640,,, --encopts=ratetol=inf:mbtree=0 --maxWidth=1920 --maxHeight=1080
```

However, if I remove `--audio-width 2=surround`, the wanted stereo mixdown for the new 2nd track works correctly, but that results in a double width surround track for the new main track which I don't want:

```
transcode-video --dry-run --avbr --target big --max-width 1920 --max-height 1080 --main-audio 2 --add-audio 1  --add-audio 3  --add-audio 12 --ac3-encoder eac3 --audio-width 1=stereo --keep-ac3-stereo --crop 264:264:2:0 /path/to/Movie.mkv 

HandBrakeCLI --input= /path/to/Movie.mkv  --output=Movie.mkv --markers --encoder=x264 --crop=264:264:2:0 --loose-anamorphic --encoder-profile=high --encoder-level=4.0 --vb=8000 --audio=2,2,1,3,12 --aencoder=eac3,ca_aac,ca_aac,ca_aac,copy --ab=640,,,, --mixdown=,stereo,,, --encopts=ratetol=inf:mbtree=0 --maxWidth=1920 --maxHeight=1080
```

I've tried changing the order of how I specify `--audio-width` (track 1 first, track 2 second, main, other) but it still ends up with the same result. Am I missing something obvious?

Thanks for looking!

**Comment by donmelton** [comment]
@Thor263 My apologies for taking so long to respond but I've been effectively AFK all day.

I'm sorry but I think your first command line is getting the results that you want (but maybe I'm misunderstanding you) ,because the second track in your output will, indeed, be stereo. That's the default mixdown for an input track that is also stereo. In other words, there's no need for the `HandBrakeCLI` command that's generated by `transcode-video` to contain `--mixdown=,stereo,,,` in it. I've optimized that away because it's redundant.

Have you examined your output with MediaInfo or another tool yet? I'm certain that your second output track will be only have two channels if the input only has two channels.

**Comment by Thor263** [comment]
@donmelton I apologize...I just realized that I failed to mention that I had expected that second track to be stereo as well, so that's why I attempted to specify it with ```--audio-width```.

Output from a few encodes:

transcode-video audio track information, using first command line, using ```--audio-width 2=surround --audio-width 1=stereo```:
```
[15:48:11]  * audio track 1
[15:48:11]    + decoder: English (DTS-HD MA) (5.1 ch) (track 2, id 0x2)
[15:48:11]      + samplerate: 48000 Hz
[15:48:11]    + mixdown: 5.1 Channels
[15:48:11]    + dither: triangular
[15:48:11]    + encoder: E-AC3 (libavcodec)
[15:48:11]      + bitrate: 640 kbps, samplerate: 48000 Hz
[15:48:11]  * audio track 2
[15:48:11]    + decoder: English (DTS-HD MA) (2.0 ch) (Lt/Rt) (track 1, id 0x1)
[15:48:11]      + samplerate: 48000 Hz
[15:48:11]    + mixdown: Dolby Surround
[15:48:11]    + dither: triangular
[15:48:11]    + encoder: AAC (Apple AudioToolbox)
[15:48:11]      + bitrate: 160 kbps, samplerate: 48000 Hz
[15:48:11]  * audio track 3
[15:48:11]    + decoder: English (DTS-HD MA) (1.0 ch) (track 3, id 0x3)
[15:48:11]      + samplerate: 48000 Hz
[15:48:11]    + mixdown: Mono
[15:48:11]    + dither: triangular
[15:48:11]    + encoder: AAC (Apple AudioToolbox)
[15:48:11]      + bitrate: 80 kbps, samplerate: 48000 Hz
```
Second command line, only specifying ```--audio-width 1=stereo``` - which appears to not set track 2 (now 3 because of the main double width) to stereo:
```
[21:58:59]  * audio track 1
[21:58:59]    + decoder: English (DTS-HD MA) (5.1 ch) (track 2, id 0x2)
[21:58:59]      + samplerate: 48000 Hz
[21:58:59]    + mixdown: 5.1 Channels
[21:58:59]    + dither: triangular
[21:58:59]    + encoder: E-AC3 (libavcodec)
[21:58:59]      + bitrate: 640 kbps, samplerate: 48000 Hz
[21:58:59]  * audio track 2
[21:58:59]    + decoder: English (DTS-HD MA) (5.1 ch) (track 2, id 0x2)
[21:58:59]      + samplerate: 48000 Hz
[21:58:59]    + mixdown: Stereo
[21:58:59]    + dither: triangular
[21:58:59]    + encoder: AAC (Apple AudioToolbox)
[21:58:59]      + bitrate: 160 kbps, samplerate: 48000 Hz
[21:58:59]  * audio track 3
[21:58:59]    + decoder: English (DTS-HD MA) (2.0 ch) (Lt/Rt) (track 1, id 0x1)
[21:58:59]      + samplerate: 48000 Hz
[21:58:59]    + mixdown: Dolby Surround
[21:58:59]    + dither: triangular
[21:58:59]    + encoder: AAC (Apple AudioToolbox)
[21:58:59]      + bitrate: 160 kbps, samplerate: 48000 Hz
[21:58:59]  * audio track 4
[21:58:59]    + decoder: English (DTS-HD MA) (1.0 ch) (track 3, id 0x3)
[21:58:59]      + samplerate: 48000 Hz
[21:58:59]    + mixdown: Mono
[21:58:59]    + dither: triangular
[21:58:59]    + encoder: AAC (Apple AudioToolbox)
[21:58:59]      + bitrate: 80 kbps, samplerate: 48000 Hz
```
I also tried a third command line without ```--audio-width``` specified and it ends up like the second command line output above. 

But...here's the Mediainfo from the first command line encode...which seems correct:
```
Audio #1
ID                                       : 2
Format                                   : E-AC-3
Format/Info                              : Enhanced AC-3
Commercial name                          : Dolby Digital Plus
Codec ID                                 : A_EAC3
Duration                                 : 2 h 1 min
Bit rate mode                            : Constant
Bit rate                                 : 640 kb/s
Channel(s)                               : 6 channels
Channel layout                           : L R C LFE Ls Rs
Sampling rate                            : 48.0 kHz
Frame rate                               : 31.250 FPS (1536 SPF)
Compression mode                         : Lossy
Delay relative to video                  : -5 ms
Stream size                              : 555 MiB (7%)
Title                                    : Surround
Language                                 : English
Service kind                             : Complete Main
Default                                  : Yes
Forced                                   : No

Audio #2
ID                                       : 3
Format                                   : AAC LC
Format/Info                              : Advanced Audio Codec Low Complexity
Codec ID                                 : A_AAC-2
Duration                                 : 2 h 1 min
Channel(s)                               : 2 channels
Channel layout                           : L R
Sampling rate                            : 48.0 kHz
Frame rate                               : 46.875 FPS (1024 SPF)
Compression mode                         : Lossy
Delay relative to video                  : -44 ms
Title                                    : Stereo
Language                                 : English
Default                                  : No
Forced                                   : No

Audio #3
ID                                       : 4
Format                                   : AAC LC
Format/Info                              : Advanced Audio Codec Low Complexity
Codec ID                                 : A_AAC-2
Duration                                 : 2 h 1 min
Channel(s)                               : 1 channel
Channel layout                           : C
Sampling rate                            : 48.0 kHz
Frame rate                               : 46.875 FPS (1024 SPF)
Compression mode                         : Lossy
Delay relative to video                  : -44 ms
Title                                    : Mono
Language                                 : English
Default                                  : No
Forced                                   : No
```

So, maybe it's the Handbrake encode output that isn't as clear as it could be in this case?  Or, what I'm beginning to realize as I'm typing, is because the second track is specified as 2.0 Lt/Rt...so it's a surround track too?

**Comment by donmelton** [comment]
@Thor263 OK, I see what you mean now with your included diagnostic information. Thanks!

And you shouldn't worry about it. :) The default behavior of `HandBrakeCLI` is a little weird in regard to stereo input tracks on occasion. Usually they're encoded with a stereo mixdown. But sometimes they're encoded with a Dolby Surround mixdown. I forgot to mention that. My apologies.

But, here's the thing, there's no real difference between those two outputs when the input is stereo. In other words, the code in `HandBrakeCLI` can't really make a surround mix when all it has to work with is two input channels. As far as I can tell, it's not really doing anything different than a stereo mixdown.

The real question is why does `HandBrakeCLI` do this on occasion? When I first noticed this my inclination was to just force the stereo mixdown. But I discussed this with my other developers and contributors and we agreed that it's best to leave this alone and trust the HandBrake development team on it. In other words, they must be doing if for a reason. :)

Anyway, I would stick with your original and first command line and trust HandBrake.

Does that answer your question?
